### PR TITLE
Adapt check that divisor is not zero for the symbolic case

### PR DESCRIPTION
**Pull request by krtab**
Fixes: #128

### CODE REVIEW

**Comment by krtab** [comment]
3376219 is a rider to appease the CI

File: src/interpret.ml [hunk 1]
```diff
@@ -87,6 +95,10 @@ module Make (P : Interpret_intf.P) :
     let* v = select v in
     f v
 
+  let ifthenelse cond t e =
+    let* cond = select cond in
+    if cond then t () else e ()
```
Comment (zapashcanon) [inline src/interpret.ml:100]: I would prefer not to add this, could you inline it directly ?

File: src/interpret.ml [hunk 2]
```diff
@@ -150,21 +162,21 @@ module Make (P : Interpret_intf.P) :
...
         | Mul -> Choice.return @@ mul n1 n2
         | Div s -> begin
-          try
-            match s with
-            | S ->
-              let> overflow = Bool.and_ (eq n1 min_int) @@ eq n2 ~-(const 1l) in
-              if overflow then Choice.trap Integer_overflow
-              else Choice.return @@ div n1 n2
-            | U -> Choice.return @@ unsigned_div n1 n2
-          with Division_by_zero -> Choice.trap Integer_divide_by_zero
```
Comment (zapashcanon) [inline src/interpret.ml:160]: I'm surprised that we don't need to catch the exception anymore in the concrete case

File: src/interpret.ml (hunk 1 above)
Comment (zapashcanon) [reply src/interpret.ml:100]: You could use `let>` instead, which is already defined and used in this file.

File: src/interpret.ml (hunk 2 above)
Comment (krtab) [reply src/interpret.ml:160]: Select works in both cases anyway so there is now an explicit guard marking the exception raising path unreachable

**Comment by zapashcanon** [comment]
Thanks!

**Pull request merged**